_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,201 @@
+
+The `<svg>` element has a role of `img` but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `img` role includes the `<svg>` element in the accessibility tree but the element has an empty accessible name `title` attribute.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" role="img">
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 3
+
+The `svg` `<circle>` elements are included in the accessibility tree but do not have accessible names.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document">
+	<title>Yellow shapes</title>
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle role="graphics-symbol" cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle role="graphics-symbol" cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+#### Failed example 4
+
+The `<g>` elements are included in the accessibility tree with the `graphics-object` role but only one of the elements has an accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="graphics-document">
+	<title>3 shapes</title>
+	<g role="graphics-object" aria-label="Blue rectangle" transform="translate(100,100)">
+        <desc>Large blue rectangle with two other shapes inside it</desc>
+		<rect fill="blue" role="graphics-object" aria-label=" large blue rectangle" width="200" height="100" y="-100" />
+		<g role="graphics-object" transform="translate(30,-90)">
+			<rect fill="green" width="50" height="90"/>
+			<rect fill="yellow" x="5" y="5" width="40" height="30" />
+			</g>
+		 </g>
+	</svg>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+The `svg` element does not have a role of `img` `graphics-document` `graphics-object` or `graphics-symbol`. 
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Inapplicable example 2
+
+The `svg` element has an accessible name but the `aria-hidden` attribute excludes the element from the accessibility tree.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" role="img" aria-label="A yellow circle" aria-hidden="true">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Inapplicable example 3
+ 
+The `svg` element is not included in the accessibility tree because the `<title>` element is empty.
+ 
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
```

Comment:
Just to add another common situation with svg in PT PA's websites:
```html
<svg role="img" title="previous page">...</svg>
```
The W3C's HTML validator fortunately indicate error in this case.

Reply:
That should pass the rule.

Reply:
`<svg role="img" title="previous page">...</svg>` would fail the rule because the svg has no accessible name / the tittle attribute is invalid.

Reply:
`title` attribute is not defined for the SVG namespace, see https://www.w3.org/TR/SVG/struct.html#TermCoreAttribute.

So @jorgeponto's test case would fail the rule.

Reply:
@jorgeponto I have added this as a fail test case.